File: DotNet/CesiumLanguageWriter/Advanced/CesiumCartesian3ValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumCartesian3ValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="Cartesian"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumCartesian3ValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumCartesian3ValuePropertyAdaptor<TFrom> : CesiumInterpolatableWriterAdaptor<TFrom, Cartesian>
        where TFrom : class, ICesiumCartesian3ValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="Cartesian"/>.</param>
        /// <param name="writeSamplesCallback">The callback to write samples of type <see cref="Cartesian"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumCartesian3ValuePropertyAdaptor([NotNull] TFrom parent,
                                                    [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, Cartesian> writeValueCallback,
                                                    [NotNull] CesiumWriterAdaptorWriteSamplesCallback<TFrom, Cartesian> writeSamplesCallback,
                                                    [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumCartesian3ValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * Adapts a class that implements {@link ICesiumCartesian3ValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@link Cartesian} values.
 * @param <TFrom> The class that implements {@link ICesiumCartesian3ValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumCartesian3ValuePropertyAdaptor<TFrom extends ICesiumCartesian3ValuePropertyWriter> extends CesiumInterpolatableWriterAdaptor<TFrom, Cartesian> {
    /**
    * Initializes a new instance.
    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link Cartesian}.
    * @param writeSamplesCallback The callback to write samples of type {@link Cartesian}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumCartesian3ValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, Cartesian> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteSamplesCallback<TFrom, Cartesian> writeSamplesCallback, @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback);
    }
}